pkg/config/utils/endpoints.go

```diff
@@ -165,6 +165,26 @@ func GetMultipleEndpoints(c pkgconfigmodel.Reader) (map[string][]APIKeys, error)
 			Keys:              []string{c.GetString("multi_region_failover.api_key")},
 		}}
 	}
+
+	// Populate preaggregation endpoint (only if unique)
+	//
+	// TODO(?): This assumes we wouldn't have a unique preaggregation API key
+	// without a unique preaggregation URL, but we should validate that.
+	if c.GetBool("enable_preaggr_pipeline") {
+		preaggURL := c.GetString("preaggr_dd_url")
```

Comment:
Would it make sense to follow the pattern of MRF just above and make a namespace for config keys? 

```suggestion
	if c.GetBool("pregaggregation_pipeline.enabled") {
		preaggURL := c.GetString("pregaggregation_pipeline.dd_url")
```